source/renderer/app/components/sidebar/SidebarCategoryWrapper.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import React from 'react';
+import type { Node } from 'react';
+import { observer } from 'mobx-react';
+import { FormattedHTMLMessage } from 'react-intl';
+import { PopOver } from 'react-polymorph/lib/components/PopOver';
+import { CATEGORIES_BY_NAME } from '../../config/sidebarConfig';
+import { useDiscreetModeFeature } from '../../features/discreet-mode';
+import NotificationDot from '../widgets/notification-dot/NotificationDot';
+import { messages } from './SidebarCategoryWrapper.messages';
+import styles from './SidebarCategoryWrapper.scss';
+
+type Props = {
+  children: Node,
+  categoryName: string,
+};
+
+const SidebarCategoryWrapper = ({ children, categoryName }: Props) => {
```

Comment:
What do you think of renaming this component to `NewFeatureNudge` or similar? At first glance wasn't clear the goal of `SidebarCategoryWrapper` without reading the body function.

Also you could create a new component for discreet mode nudge to avoid unnecessary re-renders. e.g:

```
if (
    categoryName === CATEGORIES_BY_NAME.SETTINGS.name
 ) {
 return <SettingsNudge>{children}</SettingsNudge>
}

  return children; 
```

Now only `SettingsNudge` will re-render in case `isNotificationEnabled` updates.

Reply:
The `SidebarCategoryWrapper` could have other features wrapping categories, I think NewFeatureNudge would not be appropriate in that case. 

I created a new component for the `SettingsNudge`, but more general to inject context into components without making them specific to one attribute of the context.